storage/info_rule_test.go

```diff
@@ -86,3 +87,74 @@ func TestWriteReportInfoForCluster(t *testing.T) {
 		assert.Equal(t, test.err, err)
 	}
 }
+
+// TestDBStorageReadClusterListRecommendationsNoRecommendations checks that when no recommendations
+// are stored, it is an OK state
+func TestDBStorageReadClusterListRecommendationsNoRecommendationsWithVersion(t *testing.T) {
+	mockStorage, closer := ira_helpers.MustGetMockStorage(t, true)
+	defer closer()
+
+	err := mockStorage.WriteReportForCluster(
+		testdata.OrgID, testdata.ClusterName, testdata.Report0Rules, []ctypes.ReportItem{},
+		testdata.LastCheckedAt, testdata.LastCheckedAt, testdata.LastCheckedAt, 0,
+	)
+	helpers.FailOnError(t, err)
+
+	err = mockStorage.WriteRecommendationsForCluster(
+		testdata.OrgID, testdata.ClusterName, testdata.ClusterReportEmpty, RecommendationCreatedAtTimestamp,
+	)
+	helpers.FailOnError(t, err)
+
+	err = mockStorage.WriteReportInfoForCluster(
+		testdata.OrgID, testdata.ClusterName, []types.InfoItem{}, testdata.LastCheckedAt,
+	)
+	helpers.FailOnError(t, err)
+
+	res, err := mockStorage.ReadClusterListRecommendations([]string{string(testdata.ClusterName)}, testdata.OrgID)
+	helpers.FailOnError(t, err)
+
+	expectedMeta := ctypes.ClusterMetadata{Version: ""}
+
+	assert.True(t, res[testdata.ClusterName].CreatedAt.Equal(testdata.LastCheckedAt))
+	assert.Equal(t, res[testdata.ClusterName].Meta, expectedMeta)
+}
+
+// TestDBStorageReadClusterListRecommendationsWithVersion checks that a cluster with cluster_version
+// is OK
+func TestDBStorageReadClusterListRecommendationsWithVersion(t *testing.T) {
+	mockStorage, closer := ira_helpers.MustGetMockStorage(t, true)
+	defer closer()
+
+	err := mockStorage.WriteReportForCluster(
+		testdata.OrgID, testdata.ClusterName, testdata.Report0Rules, []ctypes.ReportItem{},
+		testdata.LastCheckedAt, testdata.LastCheckedAt, testdata.LastCheckedAt, 0,
+	)
+	helpers.FailOnError(t, err)
+
+	err = mockStorage.WriteRecommendationsForCluster(
+		testdata.OrgID, testdata.ClusterName, testdata.ClusterReportEmpty, RecommendationCreatedAtTimestamp,
+	)
+	helpers.FailOnError(t, err)
+
+	err = mockStorage.WriteReportInfoForCluster(
+		testdata.OrgID,
+		testdata.ClusterName,
+		[]types.InfoItem{
+			types.InfoItem{
```

Comment:
probably it's better to use: https://github.com/RedHatInsights/insights-results-aggregator/blob/dc980ed7911b893c689c9997fd11af3da9fb5f96/server/server_test.go#L66

Reply:
That function is in a different test package. Refactoring it and moving to a common place can be part of other PR (probably it should be moved to insights-results-types repository)